code/__HELPERS/icons.dm

```diff
@@ -943,6 +943,20 @@ var/global/list/friendly_animal_types = list()
 		return J
 	return 0
 
+/atom/proc/cut_overlays()
+	overlays.Cut()
+	overlays += priority_overlays
+
+/atom/proc/add_overlay(image, priority = 0, paranoid = 0)
+	if(paranoid && (image in overlays))
+		return
+	if(priority)
+		priority_overlays += image
+		overlays += image
+	else
+		overlays -= priority_overlays
+		overlays += image
+		overlays += priority_overlays

```

Comment:
also, make this proc properly handle lists, paranoid makes passing this a list untenable.

and you'll need to figure out how to handle when image is already a priority, as something else trying to add it would cause it to dropped down to non priority.